mlflow/tracking/_model_registry/utils.py

```diff
@@ -150,6 +152,19 @@ def _get_databricks_rest_store(store_uri, **_):
     return RestStore(partial(get_databricks_host_creds, store_uri))
 
 
+def _get_databricks_uc_rest_store(store_uri, **_):
+    from mlflow.exceptions import MlflowException
+    from mlflow.version import VERSION
+
+    raise MlflowException(
+        f"Detected Unity Catalog model registry URI '{mlflow.get_registry_uri()}'. "
+        f"However, the current version of the MLflow client ({VERSION}) does not support models "
+        f"in the Unity Catalog. Please upgrade to the latest version of the MLflow Python client "
```

Comment:
`Please upgrade to the latest version of the MLflow Python client` might sound misleadng. Let's say you're using version X. If you saw this message, you would think "there is a version Y that's newer than X and supports UC registry"